// @react-spectrum/toast/dist/main.js
var $0cf579c4d0619650$exports = require("./ar-AE.main.js");
var $01a5f2e754a3aca3$exports = require("./bg-BG.main.js");
var $349346d24fea9f93$exports = require("./cs-CZ.main.js");
var $494b47caf5000db5$exports = require("./da-DK.main.js");
var $42361f52619ce7b5$exports = require("./de-DE.main.js");
var $2f3eceb7f6af8880$exports = require("./el-GR.main.js");
var $1c4e755cf68da042$exports = require("./en-US.main.js");
var $36392de7ab7cf0d8$exports = require("./es-ES.main.js");
var $f8a5185ea50f4353$exports = require("./et-EE.main.js");
var $467647dbbf34963d$exports = require("./fi-FI.main.js");
var $0bfd77936feb977b$exports = require("./fr-FR.main.js");
var $8634f5ced1fe2330$exports = require("./he-IL.main.js");
var $e8561294409b4f3f$exports = require("./hr-HR.main.js");
var $1adc73e933a677bc$exports = require("./hu-HU.main.js");
var $7425b3dc28bbc3a1$exports = require("./it-IT.main.js");
var $ed1abfad0f0fe18c$exports = require("./ja-JP.main.js");
var $209b4fa3dff0b709$exports = require("./ko-KR.main.js");
var $7f853a3d76cc8c7a$exports = require("./lt-LT.main.js");
var $db6d2778aa3efd1c$exports = require("./lv-LV.main.js");
var $cbc52695be0ba010$exports = require("./nb-NO.main.js");
var $7349a16c865e9d83$exports = require("./nl-NL.main.js");
var $9ba752c7df1444c8$exports = require("./pl-PL.main.js");
var $e71cba549c8a84b2$exports = require("./pt-BR.main.js");
var $e59fbedfd0efbd85$exports = require("./pt-PT.main.js");
var $4f5a4de90ef42f83$exports = require("./ro-RO.main.js");
var $bbf3a5b56b359b1a$exports = require("./ru-RU.main.js");
var $35b1c2966d1d12e1$exports = require("./sk-SK.main.js");
var $0d987d762afd0da6$exports = require("./sl-SI.main.js");
var $bbb1b12c9819feea$exports = require("./sr-SP.main.js");
var $2e2b2ebfa5ef1eda$exports = require("./sv-SE.main.js");
var $c9e3aeb1c2d42bfd$exports = require("./tr-TR.main.js");
var $5026bf09e6af5326$exports = require("./uk-UA.main.js");
var $f17a6e030d52029e$exports = require("./zh-CN.main.js");
var $a73ba38da5e03e12$exports = require("./zh-TW.main.js");
require("./main.css");
var $9Oq2J$react = require("react");
var $9Oq2J$reactstatelytoast = require("@react-stately/toast");
var $9Oq2J$usesyncexternalstoreshimindexjs = require("use-sync-external-store/shim/index.js");
var $9Oq2J$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $9Oq2J$reactspectrumbutton = require("@react-spectrum/button");
var $9Oq2J$reactspectrumutils = require("@react-spectrum/utils");
var $9Oq2J$spectrumiconsuiCrossMedium = require("@spectrum-icons/ui/CrossMedium");
var $9Oq2J$spectrumiconsuiInfoMedium = require("@spectrum-icons/ui/InfoMedium");
var $9Oq2J$spectrumiconsuiSuccessMedium = require("@spectrum-icons/ui/SuccessMedium");
var $9Oq2J$reactariai18n = require("@react-aria/i18n");
var $9Oq2J$reactariatoast = require("@react-aria/toast");
var $9Oq2J$reactariautils = require("@react-aria/utils");
var $9Oq2J$reactspectrumprovider = require("@react-spectrum/provider");
var $9Oq2J$reactdom = require("react-dom");
var $9Oq2J$reactariafocus = require("@react-aria/focus");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ToastContainer", () => $56cdf77f3be52eaf$export$f2815235e76a62b9);
$parcel$export(module.exports, "ToastQueue", () => $56cdf77f3be52eaf$export$f1f8569633bbbec4);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $1e709ec471c88b75$exports = {};


































$1e709ec471c88b75$exports = {
    "ar-AE": $0cf579c4d0619650$exports,
    "bg-BG": $01a5f2e754a3aca3$exports,
    "cs-CZ": $349346d24fea9f93$exports,
    "da-DK": $494b47caf5000db5$exports,
    "de-DE": $42361f52619ce7b5$exports,
    "el-GR": $2f3eceb7f6af8880$exports,
    "en-US": $1c4e755cf68da042$exports,
    "es-ES": $36392de7ab7cf0d8$exports,
    "et-EE": $f8a5185ea50f4353$exports,
    "fi-FI": $467647dbbf34963d$exports,
    "fr-FR": $0bfd77936feb977b$exports,
    "he-IL": $8634f5ced1fe2330$exports,
    "hr-HR": $e8561294409b4f3f$exports,
    "hu-HU": $1adc73e933a677bc$exports,
    "it-IT": $7425b3dc28bbc3a1$exports,
    "ja-JP": $ed1abfad0f0fe18c$exports,
    "ko-KR": $209b4fa3dff0b709$exports,
    "lt-LT": $7f853a3d76cc8c7a$exports,
    "lv-LV": $db6d2778aa3efd1c$exports,
    "nb-NO": $cbc52695be0ba010$exports,
    "nl-NL": $7349a16c865e9d83$exports,
    "pl-PL": $9ba752c7df1444c8$exports,
    "pt-BR": $e71cba549c8a84b2$exports,
    "pt-PT": $e59fbedfd0efbd85$exports,
    "ro-RO": $4f5a4de90ef42f83$exports,
    "ru-RU": $bbf3a5b56b359b1a$exports,
    "sk-SK": $35b1c2966d1d12e1$exports,
    "sl-SI": $0d987d762afd0da6$exports,
    "sr-SP": $bbb1b12c9819feea$exports,
    "sv-SE": $2e2b2ebfa5ef1eda$exports,
    "tr-TR": $c9e3aeb1c2d42bfd$exports,
    "uk-UA": $5026bf09e6af5326$exports,
    "zh-CN": $f17a6e030d52029e$exports,
    "zh-TW": $a73ba38da5e03e12$exports
};



var $b6e6aa676f8f9e47$exports = {};

$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Button", () => $b6e6aa676f8f9e47$export$1db4cca5b4ade39a, (v) => $b6e6aa676f8f9e47$export$1db4cca5b4ade39a = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-ClearButton", () => $b6e6aa676f8f9e47$export$b752ce409e5660c, (v) => $b6e6aa676f8f9e47$export$b752ce409e5660c = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast", () => $b6e6aa676f8f9e47$export$c8747d73d7954b9c, (v) => $b6e6aa676f8f9e47$export$c8747d73d7954b9c = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast--info", () => $b6e6aa676f8f9e47$export$fcf916881d7253a9, (v) => $b6e6aa676f8f9e47$export$fcf916881d7253a9 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast--negative", () => $b6e6aa676f8f9e47$export$bd5a0b226e8b302, (v) => $b6e6aa676f8f9e47$export$bd5a0b226e8b302 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast--positive", () => $b6e6aa676f8f9e47$export$5c8e63d2763f4a55, (v) => $b6e6aa676f8f9e47$export$5c8e63d2763f4a55 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast-body", () => $b6e6aa676f8f9e47$export$36e75d1f579741b3, (v) => $b6e6aa676f8f9e47$export$36e75d1f579741b3 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast-buttons", () => $b6e6aa676f8f9e47$export$19218cf016f919e5, (v) => $b6e6aa676f8f9e47$export$19218cf016f919e5 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast-content", () => $b6e6aa676f8f9e47$export$454f092ca47c3462, (v) => $b6e6aa676f8f9e47$export$454f092ca47c3462 = v);
$parcel$export($b6e6aa676f8f9e47$exports, "spectrum-Toast-typeIcon", () => $b6e6aa676f8f9e47$export$8e9e50fe392680b2, (v) => $b6e6aa676f8f9e47$export$8e9e50fe392680b2 = v);
var $b6e6aa676f8f9e47$export$1db4cca5b4ade39a;
var $b6e6aa676f8f9e47$export$b752ce409e5660c;
var $b6e6aa676f8f9e47$export$c8747d73d7954b9c;
var $b6e6aa676f8f9e47$export$fcf916881d7253a9;
var $b6e6aa676f8f9e47$export$bd5a0b226e8b302;
var $b6e6aa676f8f9e47$export$5c8e63d2763f4a55;
var $b6e6aa676f8f9e47$export$36e75d1f579741b3;
var $b6e6aa676f8f9e47$export$19218cf016f919e5;
var $b6e6aa676f8f9e47$export$454f092ca47c3462;
var $b6e6aa676f8f9e47$export$8e9e50fe392680b2;
$b6e6aa676f8f9e47$export$1db4cca5b4ade39a = `O7O3BG_spectrum-Button`;
$b6e6aa676f8f9e47$export$b752ce409e5660c = `O7O3BG_spectrum-ClearButton`;
$b6e6aa676f8f9e47$export$c8747d73d7954b9c = `O7O3BG_spectrum-Toast`;
$b6e6aa676f8f9e47$export$fcf916881d7253a9 = `O7O3BG_spectrum-Toast--info`;
$b6e6aa676f8f9e47$export$bd5a0b226e8b302 = `O7O3BG_spectrum-Toast--negative`;
$b6e6aa676f8f9e47$export$5c8e63d2763f4a55 = `O7O3BG_spectrum-Toast--positive`;
$b6e6aa676f8f9e47$export$36e75d1f579741b3 = `O7O3BG_spectrum-Toast-body`;
$b6e6aa676f8f9e47$export$19218cf016f919e5 = `O7O3BG_spectrum-Toast-buttons`;
$b6e6aa676f8f9e47$export$454f092ca47c3462 = `O7O3BG_spectrum-Toast-content`;
$b6e6aa676f8f9e47$export$8e9e50fe392680b2 = `O7O3BG_spectrum-Toast-typeIcon`;



var $e20782d0579ed886$exports = {};

$parcel$export($e20782d0579ed886$exports, "fade-out", () => $e20782d0579ed886$export$f562fd61c53bdcd9, (v) => $e20782d0579ed886$export$f562fd61c53bdcd9 = v);
$parcel$export($e20782d0579ed886$exports, "focus-ring", () => $e20782d0579ed886$export$f39a09f249340e2a, (v) => $e20782d0579ed886$export$f39a09f249340e2a = v);
$parcel$export($e20782d0579ed886$exports, "react-spectrum-ToastContainer", () => $e20782d0579ed886$export$bed15bd1aa62542d, (v) => $e20782d0579ed886$export$bed15bd1aa62542d = v);
$parcel$export($e20782d0579ed886$exports, "slide-in", () => $e20782d0579ed886$export$1c220dee9e4dbe34, (v) => $e20782d0579ed886$export$1c220dee9e4dbe34 = v);
$parcel$export($e20782d0579ed886$exports, "spectrum-FocusRing-ring", () => $e20782d0579ed886$export$4109102f950813a6, (v) => $e20782d0579ed886$export$4109102f950813a6 = v);
$parcel$export($e20782d0579ed886$exports, "spectrum-FocusRing", () => $e20782d0579ed886$export$24c7f46a6e3605dd, (v) => $e20782d0579ed886$export$24c7f46a6e3605dd = v);
$parcel$export($e20782d0579ed886$exports, "spectrum-FocusRing--quiet", () => $e20782d0579ed886$export$2927016961429360, (v) => $e20782d0579ed886$export$2927016961429360 = v);
$parcel$export($e20782d0579ed886$exports, "spectrum-Toast", () => $e20782d0579ed886$export$c8747d73d7954b9c, (v) => $e20782d0579ed886$export$c8747d73d7954b9c = v);
var $e20782d0579ed886$export$f562fd61c53bdcd9;
var $e20782d0579ed886$export$f39a09f249340e2a;
var $e20782d0579ed886$export$bed15bd1aa62542d;
var $e20782d0579ed886$export$1c220dee9e4dbe34;
var $e20782d0579ed886$export$4109102f950813a6;
var $e20782d0579ed886$export$24c7f46a6e3605dd;
var $e20782d0579ed886$export$2927016961429360;
var $e20782d0579ed886$export$c8747d73d7954b9c;
$e20782d0579ed886$export$f562fd61c53bdcd9 = `xynrca_fade-out`;
$e20782d0579ed886$export$f562fd61c53bdcd9;
$e20782d0579ed886$export$f39a09f249340e2a = `xynrca_focus-ring`;
$e20782d0579ed886$export$bed15bd1aa62542d = `xynrca_react-spectrum-ToastContainer`;
$e20782d0579ed886$export$1c220dee9e4dbe34 = `xynrca_slide-in`;
$e20782d0579ed886$export$1c220dee9e4dbe34;
$e20782d0579ed886$export$4109102f950813a6 = `xynrca_spectrum-FocusRing-ring`;
$e20782d0579ed886$export$24c7f46a6e3605dd = `xynrca_spectrum-FocusRing ${$e20782d0579ed886$export$4109102f950813a6}`;
$e20782d0579ed886$export$2927016961429360 = `xynrca_spectrum-FocusRing--quiet`;
$e20782d0579ed886$export$c8747d73d7954b9c = `xynrca_spectrum-Toast ${$e20782d0579ed886$export$24c7f46a6e3605dd}`;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







const $5b9f2c86c1b1395c$export$9194c0aa0cd7a9ff = /*#__PURE__*/ (0, $9Oq2J$react.createContext)(false);
function $5b9f2c86c1b1395c$export$fb98e3a2a4cd92d7(props) {
    let { children: children, state: state } = props;
    let ref = (0, $9Oq2J$react.useRef)();
    let { regionProps: regionProps } = (0, $9Oq2J$reactariatoast.useToastRegion)(props, state, ref);
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $9Oq2J$reactariafocus.useFocusRing)();
    let contents = /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, $9Oq2J$reactspectrumprovider.Provider), {
        UNSAFE_style: {
            background: "transparent"
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement($5b9f2c86c1b1395c$export$9194c0aa0cd7a9ff.Provider, {
        value: isFocusVisible
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement("div", {
        ...(0, $9Oq2J$reactariautils.mergeProps)(regionProps, focusProps),
        ref: ref,
        "data-position": "bottom",
        "data-placement": "center",
        className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e20782d0579ed886$exports))), "react-spectrum-ToastContainer")
    }, children)));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$reactdom))).createPortal(contents, document.body);
}




const $6c5bd1047ecc310e$export$fde44257752a9f60 = {
    info: (0, ($parcel$interopDefault($9Oq2J$spectrumiconsuiInfoMedium))),
    negative: (0, ($parcel$interopDefault($9Oq2J$spectrumiconsuiAlertMedium))),
    positive: (0, ($parcel$interopDefault($9Oq2J$spectrumiconsuiSuccessMedium)))
};
function $6c5bd1047ecc310e$var$Toast(props, ref) {
    var _state_visibleToasts_;
    let { toast: { key: key, animation: animation, content: { children: children, variant: variant, actionLabel: actionLabel, onAction: onAction, shouldCloseOnAction: shouldCloseOnAction } }, state: state, ...otherProps } = props;
    let domRef = (0, $9Oq2J$reactspectrumutils.useDOMRef)(ref);
    let { closeButtonProps: closeButtonProps, titleProps: titleProps, toastProps: toastProps } = (0, $9Oq2J$reactariatoast.useToast)(props, state, domRef);
    let { styleProps: styleProps } = (0, $9Oq2J$reactspectrumutils.useStyleProps)(otherProps);
    let stringFormatter = (0, $9Oq2J$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($1e709ec471c88b75$exports))), "@react-spectrum/toast");
    let iconLabel = variant && variant !== "neutral" ? stringFormatter.format(variant) : null;
    let Icon = $6c5bd1047ecc310e$export$fde44257752a9f60[variant];
    let isFocusVisible = (0, $9Oq2J$react.useContext)((0, $5b9f2c86c1b1395c$export$9194c0aa0cd7a9ff));
    const handleAction = ()=>{
        if (onAction) onAction();
        if (shouldCloseOnAction) state.close(key);
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement("div", {
        ...styleProps,
        ...toastProps,
        ref: domRef,
        className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Toast", {
            ["spectrum-Toast--" + variant]: variant
        }, styleProps.className, (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e20782d0579ed886$exports))), "spectrum-Toast", {
            "focus-ring": props.toast.key === ((_state_visibleToasts_ = state.visibleToasts[0]) === null || _state_visibleToasts_ === void 0 ? void 0 : _state_visibleToasts_.key) && isFocusVisible
        })),
        style: {
            ...styleProps.style,
            zIndex: props.toast.priority
        },
        "data-animation": animation,
        onAnimationEnd: ()=>{
            if (animation === "exiting") state.remove(key);
        }
    }, Icon && /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement(Icon, {
        "aria-label": iconLabel,
        UNSAFE_className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Toast-typeIcon")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement("div", {
        className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Toast-body")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement("div", {
        className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Toast-content"),
        ...titleProps
    }, children), actionLabel && /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, $9Oq2J$reactspectrumbutton.Button), {
        onPress: handleAction,
        UNSAFE_className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Button"),
        variant: "secondary",
        staticColor: "white"
    }, actionLabel)), /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement("div", {
        className: (0, $9Oq2J$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b6e6aa676f8f9e47$exports))), "spectrum-Toast-buttons")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, $9Oq2J$reactspectrumbutton.ClearButton), {
        ...closeButtonProps,
        variant: "overBackground"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, ($parcel$interopDefault($9Oq2J$spectrumiconsuiCrossMedium))), null))));
}
let $6c5bd1047ecc310e$export$8d8dc7d5f743331b = /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).forwardRef($6c5bd1047ecc310e$var$Toast);





// There is a single global toast queue instance for the whole app, initialized lazily.
let $56cdf77f3be52eaf$var$globalToastQueue = null;
function $56cdf77f3be52eaf$var$getGlobalToastQueue() {
    if (!$56cdf77f3be52eaf$var$globalToastQueue) $56cdf77f3be52eaf$var$globalToastQueue = new (0, $9Oq2J$reactstatelytoast.ToastQueue)({
        maxVisibleToasts: 1,
        hasExitAnimation: true
    });
    return $56cdf77f3be52eaf$var$globalToastQueue;
}
function $56cdf77f3be52eaf$export$320311f0e4ecb3ae() {
    $56cdf77f3be52eaf$var$globalToastQueue = null;
}
let $56cdf77f3be52eaf$var$toastProviders = new Set();
let $56cdf77f3be52eaf$var$subscriptions = new Set();
function $56cdf77f3be52eaf$var$subscribe(fn) {
    $56cdf77f3be52eaf$var$subscriptions.add(fn);
    return ()=>$56cdf77f3be52eaf$var$subscriptions.delete(fn);
}
function $56cdf77f3be52eaf$var$triggerSubscriptions() {
    for (let fn of $56cdf77f3be52eaf$var$subscriptions)fn();
}
function $56cdf77f3be52eaf$var$getActiveToastContainer() {
    return $56cdf77f3be52eaf$var$toastProviders.values().next().value;
}
function $56cdf77f3be52eaf$var$useActiveToastContainer() {
    return (0, $9Oq2J$usesyncexternalstoreshimindexjs.useSyncExternalStore)($56cdf77f3be52eaf$var$subscribe, $56cdf77f3be52eaf$var$getActiveToastContainer, $56cdf77f3be52eaf$var$getActiveToastContainer);
}
function $56cdf77f3be52eaf$export$f2815235e76a62b9(props) {
    // Track all toast provider instances in a set.
    // Only the first one will actually render.
    // We use a ref to do this, since it will have a stable identity
    // over the lifetime of the component.
    let ref = (0, $9Oq2J$react.useRef)();
    // eslint-disable-next-line arrow-body-style
    (0, $9Oq2J$react.useEffect)(()=>{
        $56cdf77f3be52eaf$var$toastProviders.add(ref);
        $56cdf77f3be52eaf$var$triggerSubscriptions();
        return ()=>{
            // When this toast provider unmounts, reset all animations so that
            // when the new toast provider renders, it is seamless.
            for (let toast of $56cdf77f3be52eaf$var$getGlobalToastQueue().visibleToasts)toast.animation = null;
            // Remove this toast provider, and call subscriptions.
            // This will cause all other instances to re-render,
            // and the first one to become the new active toast provider.
            $56cdf77f3be52eaf$var$toastProviders.delete(ref);
            $56cdf77f3be52eaf$var$triggerSubscriptions();
        };
    }, []);
    // Only render if this is the active toast provider instance, and there are visible toasts.
    let activeToastContainer = $56cdf77f3be52eaf$var$useActiveToastContainer();
    let state = (0, $9Oq2J$reactstatelytoast.useToastQueue)($56cdf77f3be52eaf$var$getGlobalToastQueue());
    if (ref === activeToastContainer && state.visibleToasts.length > 0) return /*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, $5b9f2c86c1b1395c$export$fb98e3a2a4cd92d7), {
        state: state,
        ...props
    }, state.visibleToasts.map((toast)=>/*#__PURE__*/ (0, ($parcel$interopDefault($9Oq2J$react))).createElement((0, $6c5bd1047ecc310e$export$8d8dc7d5f743331b), {
            key: toast.key,
            toast: toast,
            state: state
        })));
    return null;
}
function $56cdf77f3be52eaf$var$addToast(children, variant, options = {}) {
    // Dispatch a custom event so that toasts can be intercepted and re-targeted, e.g. when inside an iframe.
    if (typeof CustomEvent !== "undefined" && typeof window !== "undefined") {
        let event = new CustomEvent("react-spectrum-toast", {
            cancelable: true,
            bubbles: true,
            detail: {
                children: children,
                variant: variant,
                options: options
            }
        });
        let shouldContinue = window.dispatchEvent(event);
        if (!shouldContinue) return;
    }
    let value = {
        children: children,
        variant: variant,
        actionLabel: options.actionLabel,
        onAction: options.onAction,
        shouldCloseOnAction: options.shouldCloseOnAction
    };
    // Minimum time of 5s from https://spectrum.adobe.com/page/toast/#Auto-dismissible
    // Actionable toasts cannot be auto dismissed. That would fail WCAG SC 2.2.1.
    // It is debatable whether non-actionable toasts would also fail.
    let timeout = options.timeout && !options.onAction ? Math.max(options.timeout, 5000) : null;
    let queue = $56cdf77f3be52eaf$var$getGlobalToastQueue();
    let key = queue.add(value, {
        priority: $56cdf77f3be52eaf$var$getPriority(variant, options),
        timeout: timeout,
        onClose: options.onClose
    });
    return ()=>queue.close(key);
}
const $56cdf77f3be52eaf$export$f1f8569633bbbec4 = {
    /** Queues a neutral toast. */ neutral (children, options = {}) {
        return $56cdf77f3be52eaf$var$addToast(children, "neutral", options);
    },
    /** Queues a positive toast. */ positive (children, options = {}) {
        return $56cdf77f3be52eaf$var$addToast(children, "positive", options);
    },
    /** Queues a negative toast. */ negative (children, options = {}) {
        return $56cdf77f3be52eaf$var$addToast(children, "negative", options);
    },
    /** Queues an informational toast. */ info (children, options = {}) {
        return $56cdf77f3be52eaf$var$addToast(children, "info", options);
    }
};
// https://spectrum.adobe.com/page/toast/#Priority-queue
// TODO: if a lower priority toast comes in, no way to know until you dismiss the higher priority one.
const $56cdf77f3be52eaf$var$VARIANT_PRIORITY = {
    negative: 10,
    positive: 3,
    info: 2,
    neutral: 1
};
function $56cdf77f3be52eaf$var$getPriority(variant, options) {
    let priority = $56cdf77f3be52eaf$var$VARIANT_PRIORITY[variant] || 1;
    if (options.onAction) priority += 4;
    return priority;
}
