// @react-aria/toast/dist/main.js
var $3ba38fcbd805da3c$exports = require("./ar-AE.main.js");
var $06073da8a2e837af$exports = require("./bg-BG.main.js");
var $f53ed7e9d51b5d55$exports = require("./cs-CZ.main.js");
var $a6b6afdb5a98f76c$exports = require("./da-DK.main.js");
var $52ad23060875775f$exports = require("./de-DE.main.js");
var $08dcc76c0e5f63e1$exports = require("./el-GR.main.js");
var $ac3044aff017d368$exports = require("./en-US.main.js");
var $0299c089dcae9213$exports = require("./es-ES.main.js");
var $6e794080ab3e230e$exports = require("./et-EE.main.js");
var $0e8addbb92e8793c$exports = require("./fi-FI.main.js");
var $ccffda58861e52b2$exports = require("./fr-FR.main.js");
var $653260ae334153b4$exports = require("./he-IL.main.js");
var $290423d4fc0aea87$exports = require("./hr-HR.main.js");
var $da81172046d02c49$exports = require("./hu-HU.main.js");
var $e473b08597cfe1ba$exports = require("./it-IT.main.js");
var $a922e1b461f17a4b$exports = require("./ja-JP.main.js");
var $be1e07b486982d9f$exports = require("./ko-KR.main.js");
var $8ce068abcf4f94c2$exports = require("./lt-LT.main.js");
var $0f0bc08f611d7d33$exports = require("./lv-LV.main.js");
var $5166d11eab257c28$exports = require("./nb-NO.main.js");
var $cb220a55cf7a5fab$exports = require("./nl-NL.main.js");
var $917614d5adb28d77$exports = require("./pl-PL.main.js");
var $23cdda07bb218e23$exports = require("./pt-BR.main.js");
var $bb1b7d8bd48e0fee$exports = require("./pt-PT.main.js");
var $969ad461d232c044$exports = require("./ro-RO.main.js");
var $fca911223f29355e$exports = require("./ru-RU.main.js");
var $19e9a28d44f83203$exports = require("./sk-SK.main.js");
var $778cdcc7dbced5c1$exports = require("./sl-SI.main.js");
var $9359f9b7eac75ac1$exports = require("./sr-SP.main.js");
var $ac7ed8cd33134ccc$exports = require("./sv-SE.main.js");
var $b55461bb5ab189cd$exports = require("./tr-TR.main.js");
var $e6a0bdd8a70a947a$exports = require("./uk-UA.main.js");
var $c2f8abea7819ef8e$exports = require("./zh-CN.main.js");
var $765deb5c0d35c4bc$exports = require("./zh-TW.main.js");
var $35jN2$react = require("react");
var $35jN2$reactariautils = require("@react-aria/utils");
var $35jN2$reactariai18n = require("@react-aria/i18n");
var $35jN2$reactariainteractions = require("@react-aria/interactions");
var $35jN2$reactarialandmark = require("@react-aria/landmark");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "useToast", () => $ec537504faac71d2$export$a407b657d3044108);
$parcel$export(module.exports, "useToastRegion", () => $eda7c6204a682dd4$export$b8cbbb20a51697de);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ var $f49b2d6c91681304$exports = {};


































$f49b2d6c91681304$exports = {
    "ar-AE": $3ba38fcbd805da3c$exports,
    "bg-BG": $06073da8a2e837af$exports,
    "cs-CZ": $f53ed7e9d51b5d55$exports,
    "da-DK": $a6b6afdb5a98f76c$exports,
    "de-DE": $52ad23060875775f$exports,
    "el-GR": $08dcc76c0e5f63e1$exports,
    "en-US": $ac3044aff017d368$exports,
    "es-ES": $0299c089dcae9213$exports,
    "et-EE": $6e794080ab3e230e$exports,
    "fi-FI": $0e8addbb92e8793c$exports,
    "fr-FR": $ccffda58861e52b2$exports,
    "he-IL": $653260ae334153b4$exports,
    "hr-HR": $290423d4fc0aea87$exports,
    "hu-HU": $da81172046d02c49$exports,
    "it-IT": $e473b08597cfe1ba$exports,
    "ja-JP": $a922e1b461f17a4b$exports,
    "ko-KR": $be1e07b486982d9f$exports,
    "lt-LT": $8ce068abcf4f94c2$exports,
    "lv-LV": $0f0bc08f611d7d33$exports,
    "nb-NO": $5166d11eab257c28$exports,
    "nl-NL": $cb220a55cf7a5fab$exports,
    "pl-PL": $917614d5adb28d77$exports,
    "pt-BR": $23cdda07bb218e23$exports,
    "pt-PT": $bb1b7d8bd48e0fee$exports,
    "ro-RO": $969ad461d232c044$exports,
    "ru-RU": $fca911223f29355e$exports,
    "sk-SK": $19e9a28d44f83203$exports,
    "sl-SI": $778cdcc7dbced5c1$exports,
    "sr-SP": $9359f9b7eac75ac1$exports,
    "sv-SE": $ac7ed8cd33134ccc$exports,
    "tr-TR": $b55461bb5ab189cd$exports,
    "uk-UA": $e6a0bdd8a70a947a$exports,
    "zh-CN": $c2f8abea7819ef8e$exports,
    "zh-TW": $765deb5c0d35c4bc$exports
};





function $ec537504faac71d2$export$a407b657d3044108(props, state, ref) {
    let { key: key, timer: timer, timeout: timeout, animation: animation } = props.toast;
    (0, $35jN2$react.useEffect)(()=>{
        if (!timer) return;
        timer.reset(timeout);
        return ()=>{
            timer.pause();
        };
    }, [
        timer,
        timeout
    ]);
    // Restore focus to the toast container on unmount.
    // If there are no more toasts, the container will be unmounted
    // and will restore focus to wherever focus was before the user
    // focused the toast region.
    let focusOnUnmount = (0, $35jN2$react.useRef)(null);
    (0, $35jN2$reactariautils.useLayoutEffect)(()=>{
        let container = ref.current.closest("[role=region]");
        return ()=>{
            if (container && container.contains(document.activeElement)) // Focus must be delayed for focus ring to appear, but we can't wait
            // until useEffect cleanup to check if focus was inside the container.
            focusOnUnmount.current = container;
        };
    }, [
        ref
    ]);
    // eslint-disable-next-line
    (0, $35jN2$react.useEffect)(()=>{
        return ()=>{
            if (focusOnUnmount.current) focusOnUnmount.current.focus();
        };
    }, [
        ref
    ]);
    let titleId = (0, $35jN2$reactariautils.useId)();
    let descriptionId = (0, $35jN2$reactariautils.useSlotId)();
    let stringFormatter = (0, $35jN2$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($f49b2d6c91681304$exports))), "@react-aria/toast");
    return {
        toastProps: {
            role: "alert",
            "aria-label": props["aria-label"],
            "aria-labelledby": props["aria-labelledby"] || titleId,
            "aria-describedby": props["aria-describedby"] || descriptionId,
            "aria-details": props["aria-details"],
            // Hide toasts that are animating out so VoiceOver doesn't announce them.
            "aria-hidden": animation === "exiting" ? "true" : undefined
        },
        titleProps: {
            id: titleId
        },
        descriptionProps: {
            id: descriptionId
        },
        closeButtonProps: {
            "aria-label": stringFormatter.format("close"),
            onPress: ()=>state.close(key)
        }
    };
}








function $eda7c6204a682dd4$export$b8cbbb20a51697de(props, state, ref) {
    let stringFormatter = (0, $35jN2$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($f49b2d6c91681304$exports))), "@react-aria/toast");
    let { landmarkProps: landmarkProps } = (0, $35jN2$reactarialandmark.useLandmark)({
        role: "region",
        "aria-label": props["aria-label"] || stringFormatter.format("notifications")
    }, ref);
    let { hoverProps: hoverProps } = (0, $35jN2$reactariainteractions.useHover)({
        onHoverStart: state.pauseAll,
        onHoverEnd: state.resumeAll
    });
    let lastFocused = (0, $35jN2$react.useRef)(null);
    let { focusWithinProps: focusWithinProps } = (0, $35jN2$reactariainteractions.useFocusWithin)({
        onFocusWithin: (e)=>{
            state.pauseAll();
            lastFocused.current = e.relatedTarget;
        },
        onBlurWithin: ()=>{
            state.resumeAll();
            lastFocused.current = null;
        }
    });
    // When the region unmounts, restore focus to the last element that had focus
    // before the user moved focus into the region.
    // TODO: handle when the element has unmounted like FocusScope does?
    // eslint-disable-next-line arrow-body-style
    (0, $35jN2$react.useEffect)(()=>{
        return ()=>{
            if (lastFocused.current && document.body.contains(lastFocused.current)) {
                if ((0, $35jN2$reactariainteractions.getInteractionModality)() === "pointer") (0, $35jN2$reactariautils.focusWithoutScrolling)(lastFocused.current);
                else lastFocused.current.focus();
            }
        };
    }, [
        ref
    ]);
    return {
        regionProps: (0, $35jN2$reactariautils.mergeProps)(landmarkProps, hoverProps, focusWithinProps, {
            tabIndex: -1,
            // Mark the toast region as a "top layer", so that it:
            //   - is not aria-hidden when opening an overlay
            //   - allows focus even outside a containing focus scope
            //   - doesn’t dismiss overlays when clicking on it, even though it is outside
            // @ts-ignore
            "data-react-aria-top-layer": true
        })
    };
}
